static/app/components/replays/breadcrumbs/timelineGaps.tsx

```diff
@@ -13,76 +14,71 @@ import {
-
-  // returns all numbers in the range, exclusive of start and inclusive of stop
-  const range = (start, stop) =>
-    Array.from({length: stop - start}, (_, i) => start + i + 1);
-
-  const gapCol: number[] = [];
-
-  const gapFrames = framesByCol.entries();
+export default function TimelineGaps({durationMs, startTimestampMs, frames}: Props) {
+  const ranges: Array<{left: string; width: string}> = [];
+  const gapFrames = frames.entries();
   let currFrame = gapFrames.next();
 
   while (!currFrame.done) {
     let start = -1;
-    let end = -1;
+    // if no foreground frame is found, it means the gap continues to end of replay
+    let end = durationMs;
 
     // iterate through all frames until we have a start (background frame) and end (foreground frame) of gap or no more frames
-    while ((start === -1 || end === -1) && !currFrame.done) {
-      const [column, colFrame] = currFrame.value;
-      for (const frame of colFrame) {
-        // only considered start of gap if background frame hasn't been found yet
-        if (start === -1 && isBackgroundFrame(frame)) {
-          start = column;
-        }
-        // gap only ends if background frame has been found
-        if (start !== -1 && isForegroundFrame(frame)) {
-          end = column;
-        }
+    while ((start === -1 || end === durationMs) && !currFrame.done) {
+      const [, frame] = currFrame.value;
+      // only considered start of gap if background frame hasn't been found yet
+      if (start === -1 && isBackgroundFrame(frame)) {
+        start = frame.timestampMs - startTimestampMs;
+      }
+      // gap only ends if background frame has been found
+      if (start !== -1 && isForegroundFrame(frame)) {
+        end = frame.timestampMs - startTimestampMs;
       }
       currFrame = gapFrames.next();
     }
 
-    // create gap if we found have start (background frame) and end (foreground frame)
-    if (start !== -1 && end !== -1) {
-      gapCol.push(...range(start, end));
-    }
-    // if we have start but no end, that means we have a gap until end of replay
-    if (start !== -1 && end === -1) {
-      gapCol.push(...range(start, totalColumns));
+    // create gap if we found have start (background frame) and end (foreground frame) / end of replay
+    if (start !== -1) {
+      ranges.push({
+        left: toPercent(start / durationMs),
+        width: toPercent((end - start) / durationMs),
+      });
     }
   }
 
   return (
-    <Timeline.Columns totalColumns={totalColumns} remainder={0}>
-      {gapCol.map(column => (
-        <Column key={column} column={column}>
-          <Tooltip title={t('App is suspended')} isHoverable containerDisplayMode="grid">
-            <Gap style={{width: `${markerWidth}px`}} />
-          </Tooltip>
-        </Column>
-      ))}
-    </Timeline.Columns>
+    <Fragment>
+      {ranges.map(rangeCss => {
+        return (
+          <Range key={JSON.stringify(rangeCss)} css={css``} style={rangeCss}>
```

Comment:
Empty `css` prop

```suggestion
          <Range key={JSON.stringify(rangeCss)} style={rangeCss}>
```

Reply:
Also, it likely won't have a meaningful impact here, but `JSON.stringify()` is going to be slower than string interpolation, I would create your stringed key

Reply:
with interpolation:
```suggestion
          <Range key={`${rangeCss.left]-${rangeCss,width}`} style={rangeCss}>
```